acquia/mc-cs-plugin-custom-objects
# Corrected not equal and empty filters for segment

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Yes ]
| New feature/enhancement? (use the a.x branch)      | [No ]
| Deprecations?                          | [NA ]
| BC breaks? (use the c.x branch)        | [ NA]
| Automated tests included?              | [ Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


The NOT EQUAL and Empty operators when used with custom object field filters worked the opposite of what they were intended to do. This PR Fixes that. To reproduce the problem:

1. Create a custom object `Book` with a custom field `Publisher`.
2. Create 3 custom Items: BookABC, BookXYZ, BookEmpty using this custom object with `Publisher` field values as "ABC", "XYZ", and "" respectively. (Yes, the third one is an empty value) 
3. Now create 3 contacts and link these 3 custom items with 3 different contacts.
4. Now create a segment with `Not Equals` operator on `Publisher`. Provide the value "ABC" in the text box. <img width="1074" alt="image" src="https://github.com/acquia/mc-cs-plugin-custom-objects/assets/45942361/72acd9aa-d69b-49fe-bd1e-d312d24dbf3a">
5. Save the segment and run the command to update it. 
6. BookABC and BookEmpty are a part of the segment
7. The segment members don't make sense

#### Steps to test this PR:


1. Create a custom object `Book` with a custom field `Publisher`.
2. Create 3 custom Items: BookABC, BookXYZ, BookEmpty using this custom object with `Publisher` field values as "ABC", "XYZ", and "" respectively. (Yes, the third one is an empty value) 
3. Now create 3 contacts and link these 3 custom items with 3 different contacts.
4. Now create a segment with `Not Equals` operator on `Publisher`. Provide the value "ABC" in the text box. <img width="1074" alt="image" src="https://github.com/acquia/mc-cs-plugin-custom-objects/assets/45942361/72acd9aa-d69b-49fe-bd1e-d312d24dbf3a">
5. Save the segment and run the command to update it. 
8. Only the contacts linked to BookXYZ and BookEmpty should be members of this segment.
9. Now create a segment with `Empty` operator on `Publisher`. Save and update segment
10. Only the contact linked with BookEmpty should be a member of this segment.
